control/jet_vane_model.cc

```diff
@@ -0,0 +1,69 @@
+#include "control/jet_vane_model.hh"
+
+#include "third_party/experiments/eigen_helpers.hh"
+
+namespace jet {
+namespace control {
+
+namespace detail {
+SE3 compute_T_vane_unit_from_aerodynamic_frame() {
+  const jcc::Vec3 vane_center_of_pressure_vane_unit_frame = jcc::Vec3(0.0, -0.030, 0.0);
+
+  const SO3 R_intermediate_from_aerodynamic_frame = SO3::exp(jcc::Vec3(0.0, M_PI * 0.5, 0.0));
+  const SO3 R_vane_unit_from_intermediate = SO3::exp(jcc::Vec3(0.0, 0.0, -M_PI * 0.5));
+  const SO3 R_vane_unit_from_aerodynamic_frame = R_vane_unit_from_intermediate * R_intermediate_from_aerodynamic_frame;
+
+  const SE3 T_vane_unit_from_zero = SE3(R_vane_unit_from_aerodynamic_frame, vane_center_of_pressure_vane_unit_frame);
+  return T_vane_unit_from_zero;
+}
+}  // namespace detail
+
+namespace {
+double approximate_v_exhaust_mps(const double throttle, const JetConfiguration& params) {
+  // Matt says ~ sqrt(throttle)
+  return std::sqrt(throttle) * params.v_exhaust_full_throttle_mps;
+}
+}  // namespace
+
+Wrench vane_unit_wrench(const double commanded_servo_angle_rad,
+                        const JetStatus& jet_status,
+                        const VaneConfiguration& vane_cfg,
+                        const JetConfiguration& jet_cfg) {
+  const double v_exhaust_mps = approximate_v_exhaust_mps(jet_status.throttle, jet_cfg);
+
+  // TODO(jake, matt): Make this a jet status element, or otherwise computable
+  constexpr double SWIRL_ANGLE = 0.0;
+
+  // commanded_vane_angle_rad is the angle of the vane chord-line w.r.t the vehicle axis of symmetry
+  const double commanded_vane_angle_rad =
+      (commanded_servo_angle_rad / vane_cfg.gear_ratio_servo_from_vane) + vane_cfg.offset_vane_from_servo_rad;
+  const double angle_of_attack_rad = -commanded_vane_angle_rad + SWIRL_ANGLE;
+
+  const double v_exhaust_mps_sq = v_exhaust_mps * v_exhaust_mps;
+  const double dynamic_pressure_Pa = 0.5 * v_exhaust_mps_sq * jet_cfg.rho_exhaust_kg_p_m3;
+
+  // lift_coeff: is unitless
+  const double lift_coeff = angle_of_attack_rad * 2.0 * M_PI;
+  const double lift_force_at_vane_N = dynamic_pressure_Pa * lift_coeff * vane_cfg.vane_area_m2;
+
+  const double drag_coeff =
+      vane_cfg.cd0 + (4.0 * M_PI * M_PI * vane_cfg.k * (angle_of_attack_rad * angle_of_attack_rad));
+  const double drag_force_N = dynamic_pressure_Pa * drag_coeff * vane_cfg.vane_area_m2;
+  assert(drag_force_N >= 0.0);
+
+  const jcc::Vec2 force_exhaust_velocity_frame(drag_force_N, lift_force_at_vane_N);
+
+  // The vane "unit" frame; as described in the CAD
+  const SO2 vane_unit_from_exhaust_velocity = SO2::exp(SWIRL_ANGLE);
```

Comment:
Frame naming in this section is inconsistent
 - `vane_unit_from_exhaust_velocity` is used to transform "Exhaust Velocity Frame" to "_Vane Aerodynamic Frame_"
 - `T_vane_unit_from_zero` is used to transform "_Aerodynamic Frame_" to "Vane Unit Frame"